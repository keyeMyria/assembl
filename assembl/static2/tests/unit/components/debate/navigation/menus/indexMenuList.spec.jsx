import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import IndexMenuList from '../../../../../../js/app/components/debate/navigation/menus/indexMenuList';

describe('MenuList component', () => {
  it('should match the MenuList', () => {
    const items = [
      {
        id: 'foo',
        parentId: 'root'
      },
      {
        id: 'bar',
        parentId: 'root'
      }
    ];
    const props = {
      identifier: 'survey',
      className: 'debate-class',
      rootItem: 'root',
      items: items
    };
    const renderer = new ShallowRenderer();
    renderer.render(<IndexMenuList {...props} />);
    const rendered = renderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
});