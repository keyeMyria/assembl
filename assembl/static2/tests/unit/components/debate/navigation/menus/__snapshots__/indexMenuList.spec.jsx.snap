// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuList component should match the MenuList 1`] = `
<div
  className="debate-class"
>
  <Connect(DumbIndexMenuItem)
    identifier="survey"
    index={1}
    item={
      Object {
        "id": "foo",
        "parentId": "root",
      }
    }
    items={
      Array [
        Object {
          "id": "foo",
          "parentId": "root",
        },
        Object {
          "id": "bar",
          "parentId": "root",
        },
      ]
    }
    onChange={[Function]}
    selected={false}
  />
  <Connect(DumbIndexMenuItem)
    identifier="survey"
    index={2}
    item={
      Object {
        "id": "bar",
        "parentId": "root",
      }
    }
    items={
      Array [
        Object {
          "id": "foo",
          "parentId": "root",
        },
        Object {
          "id": "bar",
          "parentId": "root",
        },
      ]
    }
    onChange={[Function]}
    selected={false}
  />
</div>
`;
